tidy3d/components/heat/simulation.py

```diff
@@ -0,0 +1,760 @@
+        structures += list(self.scene.structures)
+
+        # construct slicing plane
+        axis, position = Box.parse_xyz_kwargs(x=x, y=y, z=z)
+        center = Box.unpop_axis(position, (0, 0), axis=axis)
+        size = Box.unpop_axis(0, (inf, inf), axis=axis)
+        plane = Box(center=center, size=size)
+
+        # get boundary conditions in the plane
+        boundaries = self._construct_heat_boundaries(
+            structures=structures,
+            plane=plane,
+            boundary_specs=self.boundary_specs,
+        )
+
+        # plot boundary conditions
+        for (bc_spec, shape) in boundaries:
+            ax = self._plot_boundary_condition(shape=shape, boundary_spec=bc_spec, ax=ax)
+
+        # clean up the axis display
+        axis, position = Box.parse_xyz_kwargs(x=x, y=y, z=z)
+        ax = self.scene.add_ax_labels_lims(axis=axis, ax=ax)
+        ax.set_title(f"cross section at {'xyz'[axis]}={position:.2f}")
+
+        # pylint: disable=protected-access
+        ax = self.scene._set_plot_bounds(ax=ax, x=x, y=y, z=z)
+
+        return ax
+
+    def _get_bc_plot_params(self, boundary_spec: HeatBoundarySpec) -> PlotParams:
+        """Constructs the plot parameters for given boundary conditions."""
+
+        plot_params = plot_params_heat_bc
+        condition = boundary_spec.condition
+
+        if isinstance(condition, TemperatureBC):
+            plot_params = plot_params.updated_copy(edgecolor=HEAT_BC_COLOR_TEMPERATURE)
+        elif isinstance(condition, HeatFluxBC):
+            plot_params = plot_params.updated_copy(edgecolor=HEAT_BC_COLOR_FLUX)
+        elif isinstance(condition, ConvectionBC):
+            plot_params = plot_params.updated_copy(edgecolor=HEAT_BC_COLOR_CONVECTION)
+
+        return plot_params
+
+    def _plot_boundary_condition(
+        self, shape: Shapely, boundary_spec: HeatBoundarySpec, ax: Ax
+    ) -> Ax:
+        """Plot a structure's cross section shape for a given boundary condition."""
+        plot_params_bc = self._get_bc_plot_params(boundary_spec=boundary_spec)
+        ax = self.plot_line(line=shape, plot_params=plot_params_bc, ax=ax)
+        return ax
+
+    # FIXME: probably needs revision
+    def plot_line(self, line: Shapely, plot_params: PlotParams, ax: Ax) -> Ax:
+        """Defines how a line is plotted on a matplotlib axes."""
+
+        if isinstance(line, (MultiLineString, GeometryCollection)):
+            lines = line.geoms
+        elif isinstance(line, LineString):
+            lines = [line]
+
+        for piece in lines:
+            plot_line(
+                piece,
+                ax=ax,
+                add_points=False,
+                color=plot_params.edgecolor,
+                linewidth=plot_params.linewidth,
+            )
+            # ax.add_artist(patch)
+        return ax
+
+    # pylint: disable=too-many-locals,too-many-branches,too-many-statements
+    @staticmethod
+    def _construct_heat_boundaries(
```

Comment:
this function is extremely long.. does it make sense to try to break it up into sub functions? It would be a lot of work and potentially more boilerplate / functions being used once but could make it easier to read and potentially more re-usable. I'll leave it up to you.